src/modules/adsupport.js

```diff
@@ -189,7 +189,7 @@ export default function (playerInstance, options) {
                 playerInstance.domRef.player.addEventListener('ended', playerInstance.onVastAdEnded);
 
             } else {
-                playerInstance.loadVpaid(adListId, selectedMediaFile.src);
+                playerInstance.loadVpaid(adListId, selectedMediaFile.src, videoPlayerTimeUpdate);
```

Comment:
The videoPlayerTimeUpdate function seems to have a bit of a specific use-case. It unregisters timeupdate event, for instance apart from scheduling tracking event. Not sure in vast module it's great to use this as it may end up creating weird edge-cases.

Reply:
The problem that passing the `videoPlayerTimeUpdate` to the vast module is trying to solve is that `playerInstance.domRef.player.currentTime` L209 was not being updated when loading the VPAID ad, and passing it to `loadVpaid` enabled it to be manually called by integrating with the ad interface.